pkg/registration/spoke/registration/creating_controller.go

```diff
@@ -148,3 +170,46 @@ func ClientConfigDecorator(externalServerURLs []string, caBundle []byte) Managed
 		return cluster
 	}
 }
+
+// equalClientConfigs compares two ClientConfig slices for equality
+func equalClientConfigs(configs1, configs2 []clusterv1.ClientConfig) bool {
```

Comment:
same here, clientconfigs can also be updated by user, so we cannot update clientConfigs based on equality.